syntax = "proto3";

message RegisterDevice       { string device_uuid = 1; string replica_id = 2; }
message DrainEvent           { string device_uuid = 1; }
message ReplicaFailEvent     { string  replica_id = 1; }
message MonitoredFailEvent   { string  device_uuid = 1; }
message ChangeTrConfigEvent  { string  tr_config_type = 1; string  tr_config_value = 2; }

message EventEnvelope {
  uint64  seq      = 1;
  uint64  ts_nanos = 2;
  oneof body {
    RegisterDevice       register_device   = 9;
    DrainEvent           drain             = 10;
    ReplicaFailEvent     replica_fail      = 11;
    MonitoredFailEvent   monitored_fail    = 12;
    ChangeTrConfigEvent  change_tr_config  = 13;
  }
}

message DeviceReplicaMapEntry {
  string device_uuid = 1;
  int32  replica_id   = 2;
}